homeassistant/components/dialogflow/__init__.py

```diff
@@ -84,23 +87,45 @@ class DialogFlowError(HomeAssistantError):
 
 def dialogflow_error_response(message, error):
     """Return a response saying the error message."""
-    dialogflow_response = DialogflowResponse(message["result"]["parameters"])
+    _api_version = get_api_version(message)
+    if _api_version is V1:
+        parameters = message["result"]["parameters"]
+    elif _api_version is V2:
+        parameters = message["queryResult"]["parameters"]
+    dialogflow_response = DialogflowResponse(parameters, get_api_version(message))
```

Comment:
You should be able to use `_api_version` here ?

Reply:
Good eye! Should be fixed now.